src/Racing.js

```diff
@@ -0,0 +1,31 @@
+const { maxByPosition, filterByPosition } = require("./cars");
+
+class Racing {
+  #cars;
+  #attemptCount;
+
+  constructor(cars, attemptCount) {
+    this.#cars = cars;
+    this.#attemptCount = attemptCount;
+  }
+
+  race(condition) {
+    const results = [];
+    for (let index = 0; index < this.#attemptCount.value; index++) {
+      results.push(this.#cars.map((it) => this.#move(it, condition)));
+    }
+    return results;
```

Comment:
아래와 같이 한 줄로 개선할 수 있습니다!
[Airbnb 컨벤션](https://github.com/airbnb/javascript?tab=readme-ov-file#iterators--nope)에서 for-loop의 사용을 지양하라고 했기 때문에 아래와 같이 개선해 보면 좋을 것 같아요!

```suggestion
    const results = Array.from({ length: this.#attemptCount.value }, () =>
      this.#cars.map((it) => this.#move(it, condition))
    );
    return results;
```

- [Array.from](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/from) 정적 메서드를 이용해서 배열을 생성할 수 있습니다! 자바스크립트에서 Array.from은 이 코드의 상황과 유사한 경우에 자주 사용됩니다.

Reply:
https://dev.to/ycmjason/how-to-create-range-in-javascript-539i